cnd/src/network/mod.rs

```diff
@@ -622,20 +623,22 @@ async fn handle_request(
 }
 
 #[allow(clippy::type_complexity)]
-async fn insert_state_for_bob<AL, BL, AA, BA, DB>(
+async fn insert_state_for_bob<AL, BL, AA, BA, AI, BI, DB>(
     db: DB,
     seed: RootSeed,
     state_store: Arc<InMemoryStateStore>,
     counterparty: PeerId,
-    swap_request: Request<AL, BL, AA, BA>,
+    swap_request: Request<AL, BL, AA, BA, AI, BI>,
 ) -> anyhow::Result<()>
 where
     AL: Ledger,
     BL: Ledger,
     AA: Send + 'static,
     BA: Send + 'static,
-    DB: Save<Request<AL, BL, AA, BA>> + Save<Swap>,
-    Request<AL, BL, AA, BA>: Clone,
+    AI: Send + 'static,
+    BI: Send + 'static,
+    DB: Save<Request<AL, BL, AA, BA, AI, BI>> + Save<Swap>,
+    Request<AL, BL, AA, BA, AI, BI>: Send + 'static + Clone,
```

Comment:
Shouldn't need those as we are specifying them for `AI` and `BI` above.

```suggestion
    Request<AL, BL, AA, BA, AI, BI>: Clone,
```